qutebrowser/config/configutils.py

```diff
@@ -32,22 +32,13 @@
 from PyQt5.QtGui import QFontDatabase
 from PyQt5.QtWidgets import QApplication
 
-from qutebrowser.utils import utils, urlmatch, usertypes, qtutils
+from qutebrowser.utils import utils, urlmatch, urlutils, usertypes, qtutils
 from qutebrowser.config import configexc
 
 if TYPE_CHECKING:
     from qutebrowser.config import configdata
 
 
-def _widened_hostnames(hostname: str) -> Iterable[str]:
-    """A generator for widening string hostnames.
-
-    Ex: a.c.foo -> [a.c.foo, c.foo, foo]"""
-    while hostname:
-        yield hostname
-        hostname = hostname.partition(".")[-1]
-
-
```

Comment:
I'm moving this to `urlutils` as discussed in #6372, but `urlutils` will probably end up importing `psl` and now `psl` imports `urlutils`. Should the whole psl code live in `urlutils`?

Reply:
Argh, I didn't consider that. I'd really prefer having a separate module for the PSL code, but like you say, that'd be a trivial circular import and I can't really see a way around it. So yeah, maybe it should indeed just be in `urlutils.py`?